### ISSUE TITLE
Adding a modernized option?

### ISSUE THREAD

**Issue by Hundredlegger** [schismtracker/schismtracker#389]
I know this is probably a weird thing to ask for, but could there be some update in the future to add an option for a more modernized keyboard shortcuts as well as more features and settings like MilkyTracker?

As far as I am aware, there are no modern recreations of Impulse Tracker like there is with Fasttracker 2. Fasttracker 2 has MilkyTracker with it's default FT2 Keybindings and settings, since the default is "Fasttracker 2 mode", but it also allows you to use modernized keybindings and workflow by just toggling a single button to go from "Fasttracker 2 mode" to "Milky Tracker mode".

This is kind of what I mean by this post. Basically, to add a button to toggle between "Impulse Tracker mode"/"legacy mode" and "Schism Tracker mode"/"modern mode".

I like the fact that Impulse Tracker can be entirely navigated using only the keyboard, but the fact that you cannot navigate the entire GUI using mouse only is a huge bummer, so it would be nice if it is possible to navigate the entire thing using only keyboard or only mouse to give people the option to do so.

I understand that changing this entirely is going to alienate those that are familiar with Impulse Tracker, which is why I want to know if adding a button to toggle between this and the legacy workflow is an okay feature to request.

**Comment by mrpapersonic** [comment]
This sounds like a good idea, but I'm not too familiar with how the keybinds are set up in Schism, nor would I really know which keys to set them to...

There was another effort to add completely configurable keybinds, but I don't remember it ever getting very far.

**Comment by Hundredlegger** [comment]
Well, yes, that too, but mostly a GUI that is completely mouse navigatable as well.

See, here is my main gripe: Many trackers require you to use the mouse at some point and this is perfectly fine, but Impulse Tracker is unique in that almost the entire GUI can be entirely navigated by keyboard.

Unfortunately, the mouse is kind of not actually useful whatsoever, since the mouse is almost useless and it is almost redundant. Again, I understand that it is made to suite people that use the OG Impulse Tracker, but I just think it is really weird that this hasn't been upgraded at all from a whole decade ago.

Basically, give people the option to be able to entirely navigate the GUI using the mouse A.K.A. make the mouse as useful as the keyboard.

Again, they don't have to make this the only option, hence why I asked if there could be a button to toggle between modernized and OG workflow.

**Comment by esaruoho** [comment]
the mouse was pretty useful for drawing envelopes (volume, pan, pitch, filter). i mean, the keyboard was more precise, in a way, but the mouse was way faster. 

but having configurable shortcuts in schismtracker would be pretty amazing. what happened to that project?

**Comment by mrpapersonic** [comment]
Storlek did some work on configurable keybinds in `schism/isysev.c`, but it looks like more of a work-in-progress than anything and doesn't build with the rest of the source.

**Comment by Hundredlegger** [comment]
Yes, I understand that, but you still can't do the entire GUI with just the mouse.

I think that it would be cool if the entire GUI can be navigated with keyboard or mouse so you can have both options or just one for the entirety. (The only exception I can think of is inputting a note or something on the pattern editor.)

While it is true that I asked if this could be a feature because Milky Tracker has a toggle for modernized workflow, I think it could be implemented MUCH better in Schism Tracker.

I'm not sure if what I am asking is a tall order though, but it is something that would be awesome to have.
